### DISCUSSION TITLE
Update Postgres to a more recent major version

### DISCUSSION THREAD

**Discussion by max302** [btcpayserver/btcpayserver#2332] [Ideas / Feature Requests]
I was doing unrelated research into Postgres HA clusters, and while researching feasibility of running btcpayserver on one of those I found out that the image of Postgres used in default installs is getting EOL'd soon. 

I found that @Kukks mentioned using 9.6 in #1996 but I can't seem to find any discussion as to why the 9.6 branch of Postgres was chosen in the first place. Are there any functional reasons for using this version? NPGSQL doesn't seem to have [any specific recommendations with regards to supported versions](https://www.npgsql.org/doc/compatibility.html).

According to [Postgres' versioning policy page](https://www.postgresql.org/about/news/postgresql-132-126-1111-1016-9621-and-9525-released-2165/), 9.6 is going to be retired by November 2021.

**Comment by Kukks** [comment]
> I was doing unrelated research into Postgres HA clusters, and while researching feasibility of running btcpayserver on one of those I found out that the image of Postgres used in default installs is getting EOL'd soon.
> 
> I found that @Kukks mentioned using 9.6 in #1996 but I can't seem to find any discussion as to why the 9.6 branch of Postgres was chosen in the first place. Are there any functional reasons for using this version? NPGSQL doesn't seem to have [any specific recommendations with regards to supported versions](https://www.npgsql.org/doc/compatibility.html).
> 
> According to [Postgres' versioning policy page](https://www.postgresql.org/about/news/postgresql-132-126-1111-1016-9621-and-9525-released-2165/), 9.6 is going to be retired by November 2021.

We're aware of the EOL and discussing how to make a smooth and painless migration process. The codebase can already work with all versions of postgres.
